code/ship/ship.cpp

```diff
@@ -11399,14 +11396,6 @@ void change_ship_type(int n, int ship_type, int by_sexp)
 		sp->cmeasure_count = MAX(0, sip->cmeasure_max - (sip_orig->cmeasure_max - sp->cmeasure_count));
 	}
 
-	// avoid cases where either of these are 0
-	if (sp->current_max_speed != 0 && sip_orig->max_speed != 0) {
-		sp->current_max_speed = sip->max_speed * (sp->current_max_speed / sip_orig->max_speed);
-	}
-	else {
-		sp->current_max_speed = sip->max_speed;
-	}
-
```

Comment:
Previously the (proportional) max speed would've been maintained, but ETS settings *would not* (in `ets_init_ship` right below). Resulting in inconsistent scenarios where the max speed is higher than the current energy setting should allow.

I've decided I err on the side of resetting the speed entirely, along with energy settings, but I would be amenable to maintaining energy settings (and the max speed with it).